crypto/txscript/src/standard/multisig.rs

```diff
@@ -0,0 +1,315 @@
+use crate::opcodes::codes::{OpCheckMultiSig, OpCheckMultiSigECDSA};
+use crate::script_builder::{ScriptBuilder, ScriptBuilderError};
+use std::borrow::Borrow;
+use thiserror::Error;
+
+#[derive(Error, PartialEq, Eq, Debug, Clone)]
+pub enum Error {
+    // ErrTooManyRequiredSigs is returned from multisig_script when the
+    // specified number of required signatures is larger than the number of
+    // provided public keys.
+    #[error("too many required signatures")]
+    ErrTooManyRequiredSigs,
+    #[error(transparent)]
+    ScriptBuilderError(#[from] ScriptBuilderError),
+    #[error("provided public keys should not be empty")]
+    EmptyKeys,
+}
+pub fn multisig_redeem_script_sorted(
```

Comment:
It's better to make it a private function to avoid misuse

Reply:
The main idea behind the existence of this funtion is flexibility
for example, if you have a vector/array of 3 sorted pubkeys(or wrapper struct including them or reference on them), you can filter them somehow and provide 2 of them without extra allocations, otherwise - provide unsorted mutable slice, `multisig_redeem_script` will sort it internally